### ISSUE TITLE
Update Script 26_update_v6.11.0_v6.12.0.sql

### ISSUE THREAD

**Issue by moiri** [humdek-unibe-ch/sh-selfhelp#480]
Running the script `26_update_v6.11.0_v6.12.0.sql` logs the following errors:

```
ERROR 1418 (HY000): This function has none of DETERMINISTIC, NO SQL, or READS SQL DATA in its declaration and binary logging is enabled (you *might* want to use the less safe log_bin_trust_function_creators variable)
```

The complete log is:
```
mysql> source 26_update_v6.11.0_v6.12.0.sql
Query OK, 1 row affected (0.01 sec)
Rows matched: 1  Changed: 1  Warnings: 0

Query OK, 0 rows affected (0.04 sec)

ERROR 1418 (HY000): This function has none of DETERMINISTIC, NO SQL, or READS SQL DATA in its declaration and binary logging is enabled (you *might* want to use the less safe log_bin_trust_function_creators variable)
Query OK, 0 rows affected (0.01 sec)

ERROR 1418 (HY000): This function has none of DETERMINISTIC, NO SQL, or READS SQL DATA in its declaration and binary logging is enabled (you *might* want to use the less safe log_bin_trust_function_creators variable)
Query OK, 0 rows affected (0.01 sec)

ERROR 1418 (HY000): This function has none of DETERMINISTIC, NO SQL, or READS SQL DATA in its declaration and binary logging is enabled (you *might* want to use the less safe log_bin_trust_function_creators variable)
Query OK, 0 rows affected (0.02 sec)

Query OK, 0 rows affected (0.01 sec)

Query OK, 0 rows affected (0.01 sec)

Query OK, 0 rows affected (0.01 sec)

Query OK, 0 rows affected (0.01 sec)

Query OK, 0 rows affected (0.01 sec)

Query OK, 0 rows affected (0.01 sec)

Query OK, 0 rows affected (0.02 sec)

Query OK, 0 rows affected (0.01 sec)

Query OK, 0 rows affected (0.01 sec)

```

**Comment by moiri** [comment]
After the attempted update the page fails with loads of errors:
```
BaseDb::query_db_first: SQLSTATE[42000]: Syntax error or access violation: 1305 FUNCTION pflanzenlabor.get_page_fields_helper does not exist
Fatal error: Uncaught ErrorException: Trying to access array offset on value of type bool in /home/ubuntu/selfhelp/server/page/BasePage.php:391 Stack trace:
 #0 /home/ubuntu/selfhelp/server/page/BasePage.php(391): exception_error_handler()
 #1 /home/ubuntu/selfhelp/server/page/BasePage.php(184): BasePage->fetch_page_info()
 #2 /home/ubuntu/selfhelp/server/page/SectionPage.php(49): BasePage->__construct()
 #3 /home/ubuntu/selfhelp/Selfhelp.php(223): SectionPage->__construct()
 #4 /home/ubuntu/selfhelp/Selfhelp.php(62): Selfhelp->web_call()
 #5 /home/ubuntu/selfhelp/Selfhelp.php(42): Selfhelp->init()
 #6 /home/ubuntu/selfhelp/index.php(25): Selfhelp->__construct()
 #7 {main} thrown in /home/ubuntu/selfhelp/server/page/BasePage.php on line 391
```

**Comment by moiri** [comment]
mysql version is: `8.0.36-0ubuntu0.22.04.1`
php version is: `8.1.2-1ubuntu2.14`

**Comment by Kodzhabashev** [comment]
Can you try to run:
```
SET GLOBAL log_bin_trust_function_creators = 1;
```
and then re-execute the update script.

**Comment by moiri** [comment]
Jepp that worked, thanks. Maybe set it at the beginning of the update script and unset it afterwards?
Or maybe it would be better to use explicit declarations when creating functions:
https://dev.mysql.com/doc/refman/8.0/en/stored-programs-logging.html

**Comment by Kodzhabashev** [comment]
Fixed

**Comment by Kodzhabashev** [comment]
check the stack overflow issue

**Comment by Kodzhabashev** [comment]
Add `READS SQL DATA DETERMINISTIC` to the functions. Fixed in v6.13.0
